avif/_api.py

```diff
@@ -40,6 +40,23 @@ def parse_path(self, path):
         _succeed(lib.avifDecoderParse(self._decoder))
         return self._decoder.image
 
+    def parse_data(self, data):
+        _succeed(lib.avifDecoderSetIOMemory(self._decoder, data, len(data)))
+        _succeed(lib.avifDecoderParse(self._decoder))
+        return self._decoder.image
+
     def next_images(self):
-        while lib.avifDecoderNextImage(self._decoder) == lib.AVIF_RESULT_OK:
+        while True:
+            res = lib.avifDecoderNextImage(self._decoder)
+            if res != lib.AVIF_RESULT_OK:
+                break
             yield self._decoder.image
+
+    def nth_image(self, n):
+
+        res = lib.avifDecoderNthImage(self._decoder, n)
```

Comment:
I think this should just be `_succeed(lib.avifDecoderNthImage(self._decoder, n)` no?